libs/state/spec/component-patterns/rx-state.component.spec.ts

```diff
@@ -0,0 +1,166 @@
+  extends RxState<PrimitiveState & { strWrong: string }>
+  implements AfterViewInit {
+  strChange$ = new Subject<string>();
+  strChangeWrong$ = new Subject<string>();
+  str$ = this.select('str');
+  afterViewInit = false;
+
+  @ViewChild(RxStateGlueComponent)
+  child;
+
+  constructor() {
+    super();
+    this.connect('str', this.strChange$);
+    this.connect('strWrong', this.strChangeWrong$);
+  }
+
+  ngAfterViewInit(): void {
+    this.afterViewInit = true;
+  }
+}
+
+describe('GlueTestComponent', () => {
+  let parent: RxStateGlueContainerComponent;
+  let parentFixture: ComponentFixture<RxStateGlueContainerComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [RxStateGlueComponent, RxStateGlueContainerComponent]
+    }).compileComponents();
+    parentFixture = TestBed.createComponent(RxStateGlueContainerComponent);
+    parentFixture.detectChanges();
+
+    parent = parentFixture.componentInstance;
+  });
+
+  it('should render values in parent initial', () => {
+    parent.set(initialPrimitiveState);
+    expect(parent.get()?.str).toBe(initialPrimitiveState.str);
+  });
+
+  it('should render values changes in parent', () => {
+    parent.set(initialPrimitiveState);
+    expect(parent.get()?.str).toBe(initialPrimitiveState.str);
+    parent.set({ str: 'changeInParent' });
+    // @TODO use state checker
+    expect(parent.get()?.str).toBe('changeInParent');
+  });
+
+  it('should render values in child initial', () => {
+    parent.set(initialPrimitiveState);
+    parentFixture.detectChanges();
+    expect(parent.child.get()?.str).toBe(initialPrimitiveState.str);
+  });
+
+  it('should pass values from parent to child', () => {
+    parent.set(initialPrimitiveState);
+    parentFixture.detectChanges();
+    expect(parent.child.get()?.str).toBe(initialPrimitiveState.str);
+
+    parent.set({ str: 'newParent' });
+    parentFixture.detectChanges();
+    expect(parent.child.get()?.str).toBe('newParent');
+  });
+
+  it('should work with output initialisation', () => {
+    expect(parent.afterViewInit).toBeTruthy();
+    expect(parent.child.afterViewInit).toBeTruthy();
+
+    expect(parent.get().str).toBe(undefined);
+    expect(parent.child.get().str).toBe(initialChildState.str);
+
+    const value1FromParent = 'value1FromParent';
+    parent.set({ str: value1FromParent });
+    expect(parent.get().str).toBe(value1FromParent);
+    expect(parent.child.get().str).toBe(initialChildState.str);
+
+    parentFixture.detectChanges();
+    expect(parent.get().str).toBe(value1FromParent);
+    expect(parent.child.get().str).toBe(value1FromParent);
+  });
+
+  it('should work wrong output initialisation', () => {
+    expect(parent.afterViewInit).toBeTruthy();
+    expect(parent.child.afterViewInit).toBeTruthy();
+
+    expect(parent.get().str).toBe(undefined);
+    expect(parent.get().strWrong).toBe(initialChildState.str);
+    expect(parent.child.get().str).toBe(initialChildState.str);
+  });
```

Comment:
thought about this again. shareReplay in select() operator will still cause issues since the emitted value will we replayed after first emission. so every late subscriber gets a replayed value, which is invalid for output bindings. I guess we have to come up with a `selectCold` operator which does the same thing as select, but not replaying. We could wrap the cold one into the normal one in order to handle code duplication

Reply:
The subscriber is the output binding. 

I'm sure even if we would have multiple output bindings with the same selection we would not have a late subscriber.

But we can test it.

Would you sketch a test here roughly so I see the case to test?

Reply:
> `selectCold` :
I can't image what this should be... Can you elaborate?

Reply:
will create a test for that, yes

Reply:
IS this related to this PR?
Otherwise lets finishe this one up.

Reply:
finish it!